### DISCUSSION TITLE
identify high vs no usage from discover_commands

### DISCUSSION THREAD

**Discussion by H-d-t** [DaxStudio/DaxStudio#935] [Q&A]
hi. I've stored over some months, all the DAX being executed against my model, so i have a big table now with many rows containing DAX logged from $system.discover_commands. I want to simplify my model by decommissioning no/low usage tables, columns measures. Can this or another tool deconstruct all that DAX to identify and aggregate usage per table/column/measure? I can get excellent dependency view from  $System.DISCOVER_CALC_DEPENDENCY, but that doesnt map to actual usage / deconstructed queries.

**Comment by marcosqlbi** [comment]
There are two possible approaches.
One is to correctly parse the DAX queries, locating columns / tables / measures referenced. This way, if a measure is referenced the parser can actually retrieve the underlying tables/columns actually used. This solution requires a full DAX parser, something that only Tabular Editor 3 has (it's a huge investment).
The other is to just look for tables/columns/measures referenced in the string. By knowing the data model, this can be done in an effective way by just using string searches with regular expressions. With this approach, you can see whether a column/table is used by the query, without analyzing the internal dependencies (something you can obtain with the DISCOVER_CALC_DEPENDENCY view). 
This second option is technically feasible by implementing a specific feature in DAX Studio (or you can implement it by creating your own tool) because it does not require a parser. However, for a generic tool, the limit is how to locate the history of queries to analyze.
I think that the biggest complexity is how to handle the query history and track model changes. This is required in both approaches, the alternative is an assumption of an immutable model that can create a number of issues in the tool use.
My view is that a good feature requires a large investment, probably bigger than what is affordable by a tool that does not have many resources like DAX Studio. It could be an idea for a commercial tool, but as of today also Tabular Editor does not cover the "operational" needs of monitoring a solution.
A simpler implementation could provide a quick result, but it would be very limited in its use and it could quickly become just a gadget without much practical value.
Of course, larger sponsorships for DAX Studio (or direct code contribution, but money can pay developers) could make it happen.
